lib/fizzy/execute.hpp

```diff
@@ -17,24 +17,17 @@ namespace fizzy
 // The result of an execution.
 struct execution_result
 {
-    struct trapped_tag
-    {
-    };
-
     const bool trapped = false;
     const bool has_value = false;
     const uint64_t value = 0;
 
-    /// Creates result of successful execution without returned value.
-    constexpr execution_result() = default;
-
     constexpr execution_result(uint64_t _value) noexcept : has_value{true}, value{_value} {}
 
-    constexpr explicit execution_result(trapped_tag) noexcept : trapped{true} {}
+    constexpr explicit execution_result(bool success) noexcept : trapped{!success} {}
```

Comment:
This constructor should not be used by users. We can also add default value `bool success = true` to allow `{}` as previously. But rather single way of doing things in internal code is better.